### DISCUSSION TITLE
Displaying the Selected "Settings" File.

### DISCUSSION THREAD

**Discussion by EspressoMachineWhisperer** [artisan-roaster-scope/artisan#1158] [Q&A] [answered]
Can anyone please tell me how to display the currently operating settings file. We operate 3 very different Roasting machines on Artisan and each of these machines can have several settings files. We have installed a significant number of phidgets allowing us to record many temperatures, pressures, speeds, baffles positions etc. They will often have different Event and Alarm files for our Roasting analysis. Recently we found that Artisan has been opening up on unexpected settings files and have then experienced "cross pollination" of settings into roast files. I would like to implement a startup protocol that ensures the correct settings file is operating but cannot find it displayed anywhere. Any assistance on this matter is greatly appreciated.

**Comment by MAKOMO** [comment, marked as answer]
As with any app, settings are stored in a default location determined by the operating system. For example on some Windows systems this the Windows Registry, and on macOS this is in the users ~/Library/Preference directory. Those settings are loaded on opening Artisan and the current state is saved on app exit.

Artisan allows to export and import app settings into .aset files. This allows to quickly switch between different configuration for use cases as you describe them. Last imported settings file are listed under Recent Settings. Artisan is never loading exported artisan-settings.aset files if not explicitly requested to do so.

On loading a profile recorded with a different extra device setting then in current use requires Artisan to adjust the extra device setting for technical reasons to be able to display those profiles correctly. The information for this adjustment is taken from the .alog profile. Thus loading of profiles can change extra device setups. Earlier Artisan versions asked the user for confirmation to apply those required changes or discard some of the information contained in the profile, while newer versions make a backup of the current extra device setup, apply the adjustment, and revert to the original extra device setup on next reset (or app exit). 

So your startup protocol should be to prepare one artisan-settings.aset file per configuration and just load the one you want to work with after starting Artisan. Artisan will not be changed or overwrite those external artisan-settings.aset file if not explicitly asked to do so.

**Reply by EspressoMachineWhisperer** [reply]
Thankyou for the reply Makomo. We have had problems occur when a laptop is started on a different machine from the one it was last on. I was seeking away for the user to easily check what settings file it is running on, but as you suggest, going directly to the correct file first would be best.

**Reply by Jay-NCNY** [reply]
Could you customize the colors (or theme, or background picture) for each settings file to give a visual indication of which one is running? I believe those custom settings are included in the .aset files.

**Reply by EspressoMachineWhisperer** [reply]
Thank you for the reply Jay. Good idea. I currently only use three Roasting machines so i could at least separate them by primary colour.  One of the Roasters has 5 different files for completely different Roasting Methods but at least it is confirmation the correct machine is loaded.

**Reply by roasterdave** [reply]
Try also Jay-NCNY's suggestion to use a background logo image.  Read about the feature in this [blog post](https://artisan-roasterscope.blogspot.com/2019/11/background-images.html).  Each settings file can reference a different image stored on the computer so you will know at a glance which is loaded.  Artisan allows setting the opacity of the image to make it more or less prominent.  There is an option to hide it while roasting to reduce screen clutter.  The image could show the specific roaster or make an image from text using a website like [this one](https://www.coolgenerator.com/png-text-generator).

![image](https://user-images.githubusercontent.com/30009312/236867324-7fdbaa05-eca3-49a4-8c33-78f26825ad55.png)
![image](https://user-images.githubusercontent.com/30009312/236864197-cb9d15bb-a2c4-441d-b0f2-f1233e4a15e4.png)

**Reply by EspressoMachineWhisperer** [reply]
Thank you Roasterdave. Plenty of options now i guess.

**Comment by MAKOMO** [comment]
Totally unrelated to your question: as you seem to work with some extensive Phidget setups, would you be able to help us test if the current Continuous Build works as expected on your setup?

**Reply by EspressoMachineWhisperer** [reply]
Hi Makomo. It would be my pleasure to assist wherever i could. To give you some background- I am an Industrial Electrician in Australia with over 30 years experience in PLC programing and instrumentation in the heavy Industry, from Aluminium to food. Lots of temperature control systems. My brother and I have a business called BushandBushCoffeeSystems. We have developed a measuring tool and software for the Coffee industry that defines coffee mathematically by a series of parameters critical to Roasting and Extraction. While we are still in the startup phase we have been working on this for over 8 years, almost ready to launch. We hope to extend linkages to Artisan with our measuring system and sponsor some releases and ongoing work.  I have been using Phidgets to extract deeper information on roasting and using the virtual traces with our formulas to prove and disprove analysis. I don't have any programing skills in the computer code that you use, but i have gained a fair bit of expertise in the use of phidgets to interlink with the industrial standard signals, as well as digital IO. Anything i can do to assist i am happy to contribute. Regards Jake.

**Reply by MAKOMO** [reply]
Cool! For now testing the Continuous Build with your setups and report here things that go wrong would be most helpful to be able to kick out the next version of Artisan as soon as possible. You find the Continuous Builds here:

https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous

**Comment by MAKOMO** [comment]
@EspressoMachineWhisperer: we've just uploaded a (signed) beta of the next Artisan release for review by Artisan experts like you. 

https://github.com/artisan-roaster-scope/artisan/releases/tag/v2.8.3-beta

This version includes only a few minor new features, but lots of bug fixes. Release History:
https://github.com/artisan-roaster-scope/artisan/blob/master/wiki/ReleaseHistory.md

Maybe you find time to test this version on your setup and report back issues.

Thanks,
Marko & Dave

**Reply by EspressoMachineWhisperer** [reply]
Hi Marko & Dave.
Sorry for the delay in testing but i have been working away from the test roaster, i am often 1000km away from it.
I have run V2.8.3 on a series of different settings where we semi-automate roast through phidgets and have experienced no problems at all.
Cheers Jake.

**Reply by roasterdave** [reply]
Thanks for the report.  Glad to hear Artisan v2.8.3 is working well for you. 

There is a new feature you may find useful, available in the RC build and later.  It provides one more way to distinguish your multiple roaster setups from one another.  The roaster model and capacity are now shown below the graph area whenever there is no foreground roast profile displayed.  You can customize the roaster model on the Roast>> Properties>> Setup tab. Edit the model name the click **Save Defaults** (you may have to move the cursor to another field to activate the Save Defaults button).  Click OK to exit the dialog then Help>> Save Settings to save to an existing or new settings file.  

![image](https://github.com/artisan-roaster-scope/artisan/assets/30009312/a4e6fd0b-f673-43cb-8976-462942e6195f)
